### PR TITLE
Adding conan packaging support

### PR DESCRIPTION
**Pull request by dimi309**
Hi Nigel,

So this would be it. Right now, the master branch of my fork just keeps working like your glew repository. However I have added a "conan" directory under the "build" directory and a .travis.yml.conan file to the top directory. From within the conan directory, if you execute conan test_package on a Linux or Mac machine, the package will be built (including the extensions) and tested.

Since, as you mention in the README, the generation of extensions works best on Linux and OSX, I block building the package on Windows if the version is set to "master" in the conanfile.py and test_package/conanfile.py files. If however you set the version to an existing release, e.g. 2.0.0, the package can be built on any operating system. It downloads the pregenerated source code from sourceforge.

The .travis.yml.conan file can be used to test the package on travis-ci.org. It would have to be renamed to .travis.yml of course and the current .travis.yml would have to be renamed to something else. I have tested the master mac and linux builds on travis and everything went well. By making some modifications inside the yml file (version and a couple of commands that I indicate with comments) you can also test building released versions.

An appveyor.yml file can also be used to build released versions for windows on appveyor.

If you would like to publish the package on bintray, a CONAN_UPLOAD variable and a CONAN_REMOTES variable have to be added to the yml files. This is probably too much information and, since a "picture" is worth a thousand words, you might want to have a look at the differences between my fork's [master ](https://github.com/dimi309/glew) branch and my [2.0.0/conan](https://github.com/dimi309/glew/tree/2.0.0/conan) branch which I am already using to publish prebuilt packages on [bintray](https://bintray.com/dimi309/conan-packages/glew%3Adimi309). Here are the builds:

https://travis-ci.org/dimi309/glew/builds/247368601
https://ci.appveyor.com/project/coding3d/glew/build/1.0.2

If you have any questions, don't hesitate to contact me.

As I've mentioned, I don't mind if you reject this pull request. I just thought to show you how building and publishing glew packages with conan could be done directly from the glew repository. My interest in glew is because of a [silly project](https://github.com/dimi309/small3d) I am working on during my free time :) and I like building it with conan because it saves me a lot of time. I would be more than willing to keep maintaining the glew package independently, or even with my conan changes committed to your repository, even if you are not supporting the conan procedure yourself. On the other hand, I would also be more than willing to "hand over" the entire package to you, help you set up bintray publishing and remove my own instance. Just let me know.

By the way, if by any chance you are not interested in looking further into conan at this point, would you mind if I use the cowboy hat as a logo for the package on bintray? It might help make it more recognizable :)

### CODE REVIEW

File: build/conan/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    self.copy(pattern="*.dll", dst="bin", keep_path=False)
+                else:
+                    self.copy(pattern="*32s.lib", dst="lib", keep_path=False)
+                    self.copy(pattern="*32sd.lib", dst="lib", keep_path=False)
+            else:
+                if self.options.shared:
+                    self.copy(pattern="*32.dll.a", dst="lib", keep_path=False)
+                    self.copy(pattern="*.dll", dst="bin", keep_path=False)
+                else:
+                    self.copy(pattern="*32.a", dst="lib", keep_path=False)
```
Comment (Alxandr) [inline build/conan/conanfile.py]: This does not work on my machine. I'm getting `libglew32d.a` and `libglew32d.dll.a` in my `bin` folder, yet it's trying to copy `*32.a`.

Comment (dimi309) [inline build/conan/conanfile.py]: I'm sorry. I had neglected the mingw debug builds. The latest commit to my master and 2.0.0/conan branches fixes that.

Comment (Alxandr) [inline build/conan/conanfile.py]: Great. Is that automatically published to your artifactory repository as well? Currently I've modified files under `~/.conan` which I'm guessing is going to lead to surprises down the line :P

Comment (dimi309) [inline build/conan/conanfile.py]: Normally yes, it's published, but I am having some trouble with the latest version of conan package tools right now, so I can't put it up there yet. I have submitted an issue:

https://github.com/conan-io/conan-package-tools/issues/89

For now, if you clone my repository, check out the 2.0.0/conan branch and do
`conan export dimi309/stable`

and then build your project, using the `--build glew` flag the first time you run `conan install`, you can continue without manually modifying your .conan/data repository. As soon as the package tools issue is resolved, I will also publish the latest changes on bintray.

Comment (dimi309) [inline build/conan/conanfile.py]: Ok! The issue has been resolved and I've published the corrected package on bintray. Make sure you delete your local version to be sure you're using the latest one. Thanks again for the feedback!

Comment (Alxandr) [inline build/conan/conanfile.py]: :+1:

**Comment by dimi309** [comment]
Thank you Nigel, for accepting the pull request 😄 
If you decide to start publishing the conan package on bintray, let me know. There is no rush though. In the meantime I am publishing it on my bintray repository as I was doing before. It's good that the conan configuration is now part of the main source code repository in any case.

**Comment by nigels-com** [comment]
I'm scrambling to put a release out tomorrow for Siggraph 2017.  I do want to give conan another spin sometime soon, but I'm happy for you to keep an eye on it for now.  Cheers.

**Comment by dimi309** [comment]
Of course! I don't mind maintaining this indefinitely even. Whichever works best for the project. Good luck with Siggraph!

**Comment by nigels-com** [comment]
I notice here that we need `pyenv` for the travis build to work.  I'm not familiar with that, but I'd rather not require changes to my .bashrc to build GLEW.  Is it strictly necessary?

**Comment by dimi309** [comment]
I'm afraid so, as far as I know, since conan is written in python. Having said that, maybe @drodri or @lasote can explain it better or let us know if there is an alternative, like a self-contained distributable of conan.

**Comment by lasote** [comment]
Yes, we need `python` and `pip` installed in the machine in order to install the `conan_package_tools` that allows to build the packages for different configurations. We don't have right now a prebuilt self-contained distributable for the `conan_package_tools`.
Tell me if we could help with something. Thanks!

**Comment by nigels-com** [comment]
Ah, makes sense.  Thanks.

**Comment by lasote** [comment]
Update, I've thought about it, maybe it's not needed the pyenv if the base system contains a python and pip (or we can install them with brew). We used pyenv to have available multiple python versions for testing purposes but probably here is not needed, the default python would be enough.

**Comment by Alxandr** [comment]
There should be no need for pyenv on travis. Travis containers are thrown away after build, so there isn't an issue with "polluting" it with conan packages.

**Comment by lasote** [comment]
Yes, I agree, feel free to remove it. It should work ok.

**Pull request merged**